### PR TITLE
Mapl fixes

### PR DESCRIPTION
**Pull request by kgerheiser**
* Remove extra CMakeModule logic. This was needed for hpc-stack's FindNetCDF to support static libraries.
* Update ESMA_CMake for native Apple M1 improvements.
* Update patch for Rosetta M1 for new ESMA_CMake

### CODE REVIEW

File: var/spack/repos/jcsda-emc/packages/mapl/package.py [hunk 1]
```diff
@@ -32,18 +32,15 @@ class Mapl(CMakePackage):
...
-
-    resource(
-        name='CMakeModules',
-        git='https://github.com/NOAA-EMC/CMakeModules.git',
-        tag='v1.2.0')
+        tag='v3.13.0')
 
     # Patch to configure Apple M1 chip in x86_64 Rosetta 2 emulator mode
-    patch('esma_cmake_apple_m1.patch', when='@:2.8.1')
+    patch('esma_cmake_apple_m1_rosetta.patch')
```
Comment (climbfuji) [inline var/spack/repos/jcsda-emc/packages/mapl/package.py:43]: Can this be applied to versions earlier than 2.8.1?

Comment (kgerheiser) [reply var/spack/repos/jcsda-emc/packages/mapl/package.py:43]: The patch is for ESMA_CMake which is downloaded as a resource.

Comment (kgerheiser) [reply var/spack/repos/jcsda-emc/packages/mapl/package.py:43]: But good question. Let me try some older versions.

Comment (kgerheiser) [reply var/spack/repos/jcsda-emc/packages/mapl/package.py:43]: Seems to work fine for 2.7.3 and 2.11.0

**Comment by climbfuji** [comment]
New error:
```

1 error found in build log:
     150      ESMA_cmake/operating_system/osx_extras.cmake:48 (ecbuild_warn)
     151      ESMA_cmake/esma.cmake:109 (include)
     152      CMakeLists.txt:29 (include)
     153
     154
     155    -- Building MAPL as SHARED libraries
  >> 156    CMake Error at CMakeLists.txt:75 (find_package):
     157      Could not find a configuration file for package "GFTL" that is compatible
     158      with requested version "1.5.1".
     159
     160      The following configuration files were considered but not accepted:
     161
     162        /Users/heinzell/work/spack-stack/spack-stack-package-cleanup-20220602/envs/jedi-all-apple-clang-openmpi-try3/install/apple-clang/13.1.6/gftl-shared-1.3.0-voblygd/GFTL-1.4/cmake/GFTLConfig.cmake, version: 1.4.1

See build log for details:
  /Users/heinzell/work/spack-stack/spack-stack-package-cleanup-20220602/cache/build_stage/spack-stage-mapl-2.12.3-aztrinvcp2rgwq3fcdh6ubj2mwejdvom/spack-build-out.txt

==> Warning: Skipping build of ufs-weather-model-env-1.0.0-tgbd62dv65tiah37flbqkmyzhp5653as since mapl-2.12.3-aztrinvcp2rgwq3fcdh6ubj2mwejdvom failed
```
Seems like that version requires a newer gftl-shared? If so, this needs to be encoded in the mapl package.py.

**Comment by kgerheiser** [comment]
Is that with this PR where I updated gftl-shared (and this is a submodule)?

https://github.com/NOAA-EMC/spack-stack/pull/184/files

**Comment by climbfuji** [comment]
Yes, but the line `depends_on( gftl-shared)` needs to be modified so that package dependencies are spelled out correctly.

**Comment by kgerheiser** [comment]
Oh, I get what you mean. I'll find what version.

**Comment by mathomp4** [comment]
Note to the much-more-expert-at-Spack-than-I-am here, in my test MAPL spack package (which I hope to eventually pass up to main spack) I currently have:

```
    depends_on('gftl@1.5.1:')
    depends_on('gftl-shared')

    # There was an interface change in yaFyaml, so we need to control versions
    # MAPL2 uses old, MAPL3 uses new
    depends_on('yafyaml@1.0-beta5', when='@:2')
    depends_on('yafyaml@1.0-beta8:', when='@3:')

    # pFlogger depends on yaFyaml in the same way. MAPL 2 uses old
    # yaFyaml so we need to use old pFlogger, but MAPL3 uses new yaFyaml
    depends_on('pflogger@:1.6', when='@:2+pflogger')
    depends_on('pflogger@1.8:', when='@3:+pflogger')
```

But then, I am also hoping to pass to main spack all of Tom's GFE packages (gFTL, gFTL-shared, pFUnit, pFlogger, yaFyaml, and fArgParse) as well.

You all probably don't need to worry about MAPL 3 yet, I was just future-proofing.

**Comment by mathomp4** [comment]
One more note. If it's "easier" @tclune has now released a "meta" package for his GFE libraries here:

https://github.com/Goddard-Fortran-Ecosystem/GFE

which has consistent versions of all sublibraries.

While its development is looking forward to MAPL 3, I did create two back-releases for [GCHP 14.0 ](https://github.com/Goddard-Fortran-Ecosystem/GFE/releases/tag/v0.9.0) and [(current) MAPL 2](https://github.com/Goddard-Fortran-Ecosystem/GFE/releases/tag/v0.8.0).

I'm not sure if we'll submit GFE to mainline spack as I figure the individual libraries are a bit more "correct" for something like spack, but for many users the GFE package might be the way to go. (And I've moved Baselibs to use it.)

**Review by climbfuji — APPROVED** [review]
This works on my Mac with M1/Rosetta2.

**Comment by tclune** [comment]
Alas,  NOAA is not going to want to validate the full stack.  Though they'll have to accept most of GFE at some point in the future when MAPL adopts YAML more consistently.

**Comment by kgerheiser** [comment]
That's great @mathomp4. If you get those libraries into Spack we would use that. This build uses gftl-shared to get gftl, so I found the version that corresponded to v1.5.1.

Regarding yafyaml I see that you force MAPL to use beta-5 for v2.x. We're using 0.5.1 right now, wasn't sure if we should use the betas. And version 1.0.2 fails to build for me using GCC-11.

That GFE repo looks useful too.

**Comment by tclune** [comment]
@kgerheiser   GFE 1.0.2 should work with GCC 11.2.    There is some bug in 11.1 that was not worth working around since 11.2 was out by the time I noticed.    If this was with 11.2 or 11.3, could you open a ticket on GFE or yaFyaml with the details?  Thank.

**Comment by climbfuji** [comment]
Tom, 1.0.2 also fails for me with ***@***.***

**Comment by tclune** [comment]
@climbfuji  - I do not follow.    Which compiler + OS does it fail with, and is that a compile time error or a run time error.

**Comment by kgerheiser** [comment]
Might have been due to an old gftl-version. Using the gftl-shared we just updated to worked just now with GCC 11.3.0.

**Comment by climbfuji** [comment]
I will try again with gftl -shared 1.5.0 on macos with clang 13.1.6. If there is still a problem, I'll report the error here.

**Comment by tclune** [comment]
OK - yes  just realized I failed to update the minimum version requirements when I did the last sweep.  (So tedious ...)

If important, I can release a patch with that extra info.  Otherwise, I'll capture it in the next release.

**Comment by mathomp4** [comment]
@kgerheiser @climbfuji For MAPL 2, don't go beyond yafyaml beta5. The versions I have listed in [v0.8.0](https://github.com/Goddard-Fortran-Ecosystem/GFE/releases/tag/v0.8.0):

| Package     | Version      |
| :------     | :------      |
| gFTL        | v1.5.5       |
| gFTL-shared | v1.3.6       |
| fArgParse   | v1.1.2       |
| yaFyaml     | v1.0-beta5.1 |
| pFlogger    | v1.6.1       |
| pFUnit      | v4.2.2       |

I can at least guarantee work with "recent" MAPL. I think they even work back with 2.8.0 so that might be a good target for you to aim for...and just freeze there until MAPL 3.

ETA: the "beta5.1" was due to a CMake bug that Tom fixed in GFE 1 but had to be backported for use in "pre-historical" GFE. It had to do with building yaFyaml inside GFE rather than on its own.

**Comment by climbfuji** [comment]
@mathomp4 That's good to know, thank you!

**Comment by climbfuji** [comment]
> @mathomp4 That's good to know, thank you!

@kgerheiser I'll make this change in my next PR ...

**Comment by mathomp4** [comment]
Note of course, if you don't need some libraries, you don't need to build them. I of course build them all because we use all of them in our MAPL development. But if you aren't running unit tests, say, pFUnit isn't needed.

**Pull request merged**